src/amplitude-client.js

```diff
@@ -354,14 +354,28 @@ AmplitudeClient.prototype._migrateUnsentEvents = function _migrateUnsentEvents(c
  * @private
  */
 AmplitudeClient.prototype._trackParamsAndReferrer = function _trackParamsAndReferrer() {
+  let utmProperties;
+  let referrerProperties
+  let gclidProperties;
   if (this.options.includeUtm) {
-    this._initUtmData();
+    utmProperties = this._initUtmData();
   }
   if (this.options.includeReferrer) {
-    this._saveReferrer(this._getReferrer());
+    referrerProperties = this._saveReferrer(this._getReferrer());
   }
   if (this.options.includeGclid) {
-    this._saveGclid(this._getUrlParams());
+    gclidProperties = this._saveGclid(this._getUrlParams());
+  }
+  if (this.options.logAttributionCapturedEvent) {
+    const attributionProperties = Object.assign(
```

Comment:
I believe you could make this more concise/readable by using the `...` spread operator instead of `Object.assign`

`const attributionProperties = { ... utmProperties, ... referrerProperties, ...gclidProperties }`

Reply:
Updated!